services/dis-apim-operator/internal/config/conf.go

```diff
@@ -0,0 +1,70 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/knadh/koanf/parsers/toml/v2"
+	"github.com/knadh/koanf/providers/env"
+	"github.com/knadh/koanf/providers/file"
+	"github.com/knadh/koanf/providers/posflag"
+	"github.com/knadh/koanf/v2"
+	"github.com/spf13/pflag"
+)
+
+type AzureConfig struct {
+	SubscriptionId  string `json:"subscriptionId,omitempty" koanf:"subscriptionId" toml:"subscriptionId"`
+	ResourceGroup   string `json:"resourceGroup,omitempty" koanf:"resourceGroup" toml:"resourceGroup"`
+	ApimServiceName string `json:"apimServiceName,omitempty" koanf:"apimServiceName" toml:"apimServiceName"`
+}
+
+func LoadConfig(configFile string, flagset *pflag.FlagSet) (*AzureConfig, error) {
+	k := koanf.New(".")
+
+	// Load from file
+	if configFile != "" {
+		if _, err := os.Stat(configFile); err == nil {
+			err := k.Load(file.Provider(configFile), toml.Parser())
+			if err != nil {
+				return nil, fmt.Errorf("error loading config file: %w", err)
+			}
+		}
+	}
+
+	// Load from environment
+	err := k.Load(env.Provider("DISAPIM_", ".", func(s string) string {
+		return toCamelCase(strings.ToLower(strings.TrimPrefix(s, "DISAPIM_")))
+	}), nil)
+
+	if err != nil {
+		return nil, fmt.Errorf("error loading environment variables: %w", err)
+	}
+
+	// Load from flags
+	err = k.Load(posflag.Provider(flagset, ".", k), nil)
+	if err != nil {
+		return nil, fmt.Errorf("error loading flags: %w", err)
+	}
+	var c AzureConfig
+	if err := k.Unmarshal("", &c); err != nil {
+		return nil, fmt.Errorf("error unmarshalling config: %w", err)
+	}
+	return &c, nil
+}
+
+func LoadConfigOrDie(configFile string, flagset *pflag.FlagSet) *AzureConfig {
+	c, err := LoadConfig(configFile, flagset)
+	if err != nil {
+		panic(err)
+	}
+	return c
+}
+
+func toCamelCase(snake string) string {
```

Comment:
nit: https://pkg.go.dev/github.com/iancoleman/strcase

Reply:
but...:
https://github.com/iancoleman/strcase/issues/46
https://github.com/iancoleman/strcase/issues/51

Reply:
Fun stuff :D  - I guess pkg text could be used as well hehe, e.g
```
func toCamelCase(s string) string {
	parts := strings.Split(s, "_")
	titleCaser := cases.Title(language.Und)
	for i := 0; i < len(parts); i++ {
		parts[i] = titleCaser.String(parts[i])
	}
	return strings.Join(parts, "")
}
```

Reply:
He had a similar implementation in a previous commit > https://github.com/Altinn/altinn-platform/pull/1086/commits/59b2d1876eabe9a6f6ddbd00598fb03b51e4d9be#diff-9b660e37e4ef495a95207cfad1253f639a676ad8d871a264ba0b9cdb18b97514R63-R69

Reply:
hehe, this is fun! 
I actually had a look at both your proposals before committing this. My first implementation I was using strings.Title, but that is deprecated as it doesn't support unicodepunctiation and some language specific rules when it comes to title. 
As I in this case only really care to potentially change tree environment variables I opted to not depend on another module or use any language specific implementation (thought it might make the code slower and/or harder to follow). I just want the first letter to be upper case.
I have no strong feelings towards any solution, this is just the reasoning behind my choice.

Reply:
Deprecated, yes hehe, that's why now one needs to use x/text for Title. 
> I opted to not depend on another module or use any language specific implementation

Yeah, agree - it was mostly for fun.

Reply:
Marking this as resolved as I feel we agree on the solution. If the use case was more advanced or gets more advanced we should look at a different solution, but for now the code have testcases for all the mappings needed by the application to load config correctly. Please reopen/comment if you disagree